### PR TITLE
[extension](feature)Mysql database import doris by external tables

### PR DESCRIPTION
**Pull request by Toms1999**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Toms1999** [comment]
alter chinese

File: extension/mysql_to_doris/all_tables.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+echo "use $d_mysql; show tables;" |mysql -h$mysql  -uroot -p$mysql_password 2>/dev/null >> ./files/tables
+
+#delete tables first line
+sed -i '1d' ./files/tables
+
+#reference tables to create tables.sql
+for table in $(awk -F '\n' '{print $1}' ./files/tables)
+        do
+        sed -i "/${table}view/d" ./files/tables
+        echo "use $d_mysql; show create table ${table};" |mysql -h$mysql -uroot -p$mysql_password 2>/dev/null >> ./files/tables.sql
```
[the remark this answers is not in the document]
Comment (Toms1999) [reply extension/mysql_to_doris/all_tables.sh]: you are right，I've already fixed it

File: extension/mysql_to_doris/all_tables.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        do
+        sed -i "0,/ApacheDorisHostIp/s/ApacheDorisHostIp/${mysql}/" ./files/tables.sql
+        sed -i "0,/ApacheDorisHostPassword/s/ApacheDorisHostPassword/${mysql_password}/" ./files/tables.sql
+        sed -i "0,/ApacheDorisDataBases/s/ApacheDorisDataBases/${d_mysql}/" ./files/tables.sql
+        sed -i "0,/ApacheDorisTables/s/ApacheDorisTables/${t_name}/" ./files/tables.sql
+
+done
+######################################################################################################################################################################
+#replace mysql type with doris
+sed -i 's/text/string/g' ./files/tables.sql
```
[the remark this answers is not in the document]
Comment (Toms1999) [reply extension/mysql_to_doris/all_tables.sh]: i will string replace with varchar(65535),but some data type not a good idea,I hope to improve it in the future by issue

File: extension/mysql_to_doris/all_tables.sh [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+sed -i '/UNIQUE KEY/d' ./files/tables.sql
+#delete , at the beginning (
+sed -i '/,\s*$/{:loop; N; /,\(\s*\|\n\))/! bloop; s/,\s*[\n]\?\s*)/\n)/}' ./files/tables.sql
+
+#delete a line on keyword
+sed -i -e '$!N;/\n.*ENGINE=MYSQL/!P;D' ./files/tables.sql
+#replace mysql password、database、table、host
+for t_name in $(awk -F '\n' '{print $1}' ./files/tables)
+        do
+        sed -i "0,/ApacheDorisHostIp/s/ApacheDorisHostIp/${mysql}/" ./files/tables.sql
```
Comment (caoliang-web) [inline extension/mysql_to_doris/all_tables.sh:89]: It is incorrect to use ${mysql} here, it should be ${mysql_host}

Comment (Toms1999) [reply extension/mysql_to_doris/all_tables.sh]: I have completed the repair and test once

File: extension/mysql_to_doris/all_tables.sh [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+sed -i 's/DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP//g' ./files/tables.sql
+sed -i 's/DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP//g' ./files/tables.sql
+sed -i 's/CHARACTER SET utf8mb4 COLLATE utf8mb4_bin//g' ./files/tables.sql
+sed -i 's/DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP//g' ./files/tables.sql
+sed -i 's/DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP//g' ./files/tables.sql
+sed -i 's/CHARACTER SET utf8//g' ./files/tables.sql
+sed -i 's/COLLATE utf8mb4_general_ci//g' ./files/tables.sql
+sed -i 's/COLLATE utf8_general_ci//g'  ./files/tables.sql
+sed -i 's/COLLATE utf8_general_ci//g'  ./files/tables.sql
+sed -i 's/\<tinytext\>/varchar(65535)/g' ./files/tables.sql
```
Comment (caoliang-web) [inline extension/mysql_to_doris/all_tables.sh:113]: The range of doris varchar is 1-65533, which needs to be modified to 65533

Comment (Toms1999) [reply extension/mysql_to_doris/all_tables.sh]: i has fixed ,thanks

File: extension/mysql_to_doris/user_define_tables.sh [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+sed -i 's/varbinary([0-9][0-9][0-9])/varchar(65533)/g' ./user_files/tables.sql
+sed -i 's/string([0-9][0-9][0-9])/varchar(65533)/g' ./user_files/tables.sql
+sed -i 's/CHARACTER SET utf8mb4 COLLATE utf8mb4_bin//g' ./user_files/tables.sql
+
+
+#######################################
+#import doris
+for table in $(awk -F '\n' '{print $1}' ./conf/tables)
+        do
+        echo "use $d_doris; drop table if exists ${table};" |mysql -h$master_host -P$master_port -uroot -p$doris_password 2>/dev/null
```
Comment (caoliang-web) [inline extension/mysql_to_doris/user_define_tables.sh:139]: for loop not ending

Comment (Toms1999) [reply extension/mysql_to_doris/user_define_tables.sh:139]: the for loop read result set is limited，so it is auto over

File: extension/mysql_to_doris/conf/tables [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+####################################################################
+# The tables is used to define tables
+####################################################################
```
Comment (caoliang-web) [inline extension/mysql_to_doris/conf/tables:22]: This file is not a shell script. When the user_define_tables.sh script is used to read ./conf/tables, the above contents will be read, resulting in the wrong table name being obtained and the script execution failing. These contents need to be removed.

Comment (Toms1999) [reply extension/mysql_to_doris/conf/tables]: delete all content

**Review by caoliang-web — DISMISSED** [review]
LGTM

**Review by hf200012 — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
hi @Toms1999 , you need to add `extension/mysql_to_doris/conf/tables` to the `.licenserc.yaml` to pass the license check

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**